src/gui/notifications/dbusnotifier.cpp

```diff
@@ -0,0 +1,224 @@
+/*
+ * Bittorrent Client using Qt4/Qt5 and libtorrent.
+ * Copyright (C) 2015
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * In addition, as a special exception, the copyright holders give permission to
+ * link this program with the OpenSSL project's "OpenSSL" library (or with
+ * modified versions of it that use the same license as the "OpenSSL" library),
+ * and distribute the linked executables. You must obey the GNU General Public
+ * License in all respects for all of the code used other than "OpenSSL".  If you
+ * modify file(s), you may extend this exception to your version of the file(s),
+ * but you are not obligated to do so. If you do not wish to do so, delete this
+ * exception statement from your version.
+ *
+ */
+
+#include "dbusnotifier.h"
+#include "notifications.h"
+#include <QDesktopServices>
+#include <QDir>
+#include <QUrl>
+#include <QWidget>
+#include <QtDBus/QDBusConnection>
+#include <QtCore/QDebug>
+
+namespace {
+    static const QLatin1String ACTION_NAME_DEFAULT("activate-context");
+    static const QLatin1String ACTION_NAME_OPEN_FINISHED_TORRENT("document-open"); // it must be named as the corresponding FDO icon
+    static const QLatin1String DBUS_SERVICE_NAME("org.freedesktop.Notifications");
+    static const QLatin1String DBUS_INTERFACE_NAME("org.freedesktop.Notifications");
+    static const QLatin1String DBUS_PATH("/org/freedesktop/Notifications");
+}
```

Comment:
in C++03, as I understand, 
```c++ 
namespace {}
``` 
is by definition equal to 
```c++
namespace RandomUniqueName {}
using namespace RandomUniqueName;
```
But symbols from it still can be used where external linkage is needed.
Here are more details: http://www.comeaucomputing.com/techtalk/#nostatic